db/commitlog/commitlog.cc

```diff
@@ -124,35 +124,38 @@ db::commitlog::descriptor::descriptor(replay_position p, const std::string& fnam
         : descriptor(p.id, fname_prefix) {
 }
 
+const std::string db::commitlog::descriptor::SEPARATOR("-");
+const std::string db::commitlog::descriptor::FILENAME_PREFIX("CommitLog" + SEPARATOR);
+const std::string db::commitlog::descriptor::FILENAME_EXTENSION(".log");
 
-db::commitlog::descriptor::descriptor(const sstring& filename, const std::string& fname_prefix)
-        : descriptor([&filename, &fname_prefix]() {
-            std::smatch m;
-            // match both legacy and new version of commitlogs Ex: CommitLog-12345.log and CommitLog-4-12345.log.
-                std::regex rx("(?:Recycled-)?" + fname_prefix + "((\\d+)(" + SEPARATOR + "\\d+)?)" + FILENAME_EXTENSION);
-                std::string sfilename = filename;
-                auto cbegin = sfilename.cbegin();
-                // skip the leading path
-                // Note: we're using rfind rather than the regex above
-                // since it may run out of stack in debug builds.
-                // See https://github.com/scylladb/scylla/issues/4464
-                auto pos = std::string(filename).rfind('/');
-                if (pos != std::string::npos) {
-                    cbegin += pos + 1;
-                }
-                if (!std::regex_match(cbegin, sfilename.cend(), m, rx)) {
-                    throw std::domain_error("Cannot parse the version of the file: " + filename);
-                }
-                if (m[3].length() == 0) {
-                    // CMH. Can most likely ignore this
-                    throw std::domain_error("Commitlog segment is too old to open; upgrade to 1.2.5+ first");
-                }
+static const std::regex allowed_prefix("[a-zA-Z]+" + db::commitlog::descriptor::SEPARATOR);
```

Comment:
Nitpick about the commit message: the keyword "static" here means "file local", not shared. It would have been shared even without this "static" keyword. But I agree "static" (file local) is good here as well.

Reply:
Shared between descriptors. Not compilation units.

Reply:
And descriptor creation is unit tested (though indirectly).

Reply:
Yes, of course, I was just pointing out that you used the word "static" in the commit message which doesn't refer to the "static" in this line but a different concept of "static". Nevermind, it's just a silly nitpick on the commit message. You can ignore it.

What do you mean when you say it's unit-tested but indirectly? I'm sure it's used in every one of our other functional tests which indeed create commit logs, but do those tests use different pathnames and different prefix and old and new versions and so on? If they don't, it is possible that this patch broke the regular expression in one of these obscure cases and we won't even know. Of course, I'm not demanding that it should be unit-tested right now - this is pre-existing code after all.